app/services/research_safety_adapters/scishield_training_synchronizer.rb

```diff
@@ -12,20 +12,40 @@ def synchronize
         Rails.logger.error(msg)
         Rollbar.error(msg) if defined?(Rollbar)
       else
+        user_certs = {}
+
+        # In some cases, making many requests in a row results in the API not
+        # responding. Making requests in batches with sleep in between seems
+        # to make the API work in cases where this is a problem. `sleep` is put
+        # first here because `api_unavailable?` will have already made 10 API
+        # requests.
+        users.in_batches(of: batch_size) do |user_batch|
+          sleep(batch_sleep_time)
+
+          user_batch.each do |user|
+            adapter = ScishieldApiAdapter.new(user, api_client)
+            begin
+              cert_names = adapter.certified_course_names_from_api
+              user_certs[user.id.to_s] = cert_names if cert_names.presence
+            rescue => e
+              Rails.logger.error(e.message)
+              Rollbar.error(e.message) if defined?(Rollbar)
+            end
+          end
+        end
+
         ScishieldTraining.transaction do
-          ScishieldTraining.delete_all
+          ScishieldTraining.delete(user_certs.keys.map(&:to_s))
 
-          users.find_each do |user|
-            adapter = ScishieldApiAdapter.new(user)
-            cert_names = adapter.certified_course_names_from_api
+          user_certs.each do |user_id, cert_names|
             trainings_added = 0
 
-            cert_names.each do |cert_name|
-              st = ScishieldTraining.create(user_id: user.id, course_name: cert_name)
+            cert_names.each do |name|
+              st = ScishieldTraining.create(user_id:, course_name: name)
```

Comment:
I'm not sure what would cause this to fail, but I think we should raise in that case:
```suggestion
              st = ScishieldTraining.create!(user_id:, course_name: name)
```